spinnman/processes/get_machine_process.py

```diff
@@ -99,22 +111,22 @@ def _make_chip(self, chip_info, machine):
 
         # Create the chip
         return Chip(
-            x=chip_info.x, y=chip_info.y, processors=processors,
+            x=chip_info.x, y=chip_info.y, n_processors=n_cores,
             router=router, sdram=sdram,
             ip_address=chip_info.ethernet_ip_address,
             nearest_ethernet_x=chip_info.nearest_ethernet_x,
-            nearest_ethernet_y=chip_info.nearest_ethernet_y)
+            nearest_ethernet_y=chip_info.nearest_ethernet_y,
+            down_cores=down_cores)
 
     def _make_router(self, chip_info, machine):
         links = list()
         for link in chip_info.working_links:
-            dest = machine.xy_over_link(chip_info.x, chip_info.y, link)
-            if ((chip_info.x, chip_info.y, link) not in self._ignore_links
-                    and dest not in self._ignore_chips
-                    and dest in self._chip_info):
-                dest_x, dest_y = dest
+            dest_xy = machine.xy_over_link(
+                chip_info.x, chip_info.y, link)
```

Comment:
no need of the split

Reply:
Yes there is because it needs a triple (x, y, p)  and the pair (ChipInfo, P)

Reply:
no im on about :

Reply:
dest_xy = machine.xy_over_link(
                chip_info.x, chip_info.y, link)

Reply:
yep the method takes 3 int params and not a ChipInfo OBJECT and and int so you NEED the split!

Reply:
*sighs* split meant over 2 lines. it can fit in 80 chars without the LINE SPLIT

Reply:
ah that kind of split! Agreed. Fixed

Did so many fixes of tupling and untupling that i read it that way.

Reply:
im still inclined to my view that this whole dead chip / core / ipaddress should be some object instead of a set of tuples with differing sizes. 

class dead_core_store()

def __init__(self, x, y, p=None, ip_address=None):
etc etc

properties etc

then youd not need all those ifs and lens, you jsut take the property of the object....

Reply:
I had wondered the same thing myself so Objects here we come.